node/peers.js

```diff
@@ -136,12 +136,6 @@ TChannelPeers.prototype.delete = function del(hostPort) {
     var self = this;
     var peer = self._map[hostPort];
     delete self._map[hostPort];
-    if (self.subChannels) {
```

Comment:
This should be corrected not removed.  The invariant we want is that if a peer is in a sub channel, then it's on the top channel too.  So when you remove a peer from a top channel, it must be removed from all sub channels.

The clear method is even more/differently wrong in that it doesn't have any affordance.